src/ext/isearch.lisp

```diff
@@ -383,9 +395,8 @@
         (concatenate 'string
                      *isearch-string*
                      (string c)))
-  (with-point ((start-point (current-point)))
-    (unless (funcall *isearch-search-function* (current-point) *isearch-string*)
-      (move-point (current-point) start-point)))
+  (unless (funcall *isearch-search-function* (current-point) *isearch-string*)
+    (move-point (current-point) *isearch-start-point*))
```

Comment:
I think the current behavior is not intended that the cursor point doesn't go back when the search fails.

Reply:
Ah, I see.
if you write the following, the current-point might not be broken when interrupted.

```common-lisp
(with-point ((point (current-point)))
  (when (funcall *isearch-search-function* point *isearch-string*)
    (move-point (current-point) point)))
```

Reply:
Oh, that looks better. I'll rewrite it as you suggested. Thanks!

Reply:
Sorry, this has to be the code I originally changed.

For example, when I have a buffer like the following:

```
[a]pple orange grape
``` 

When starting ISearch with `C-s lisp`, then the cursor moves to `app[l]e` even though the search is failed.

It's better to go back to the starting point when the search has failed. How do you think?

Reply:
PR #1002